### PR TITLE
Steady hand lasers

### PR DESCRIPTION
**Pull request by sethalves**
- fix-up parenting code in Line3dOverlay
- handControllerGrab's hand lasers are now children of controller joints for smoother motion

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- use hand controllers
- far grab things
- use hand lasers on tablet and web entities
- hand lasers should be less jittery than in master, but there should be no other changes

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8893c4770e2ce5cc022d980950b826adfba4cf3d available at: https://gist.github.com/177554542bd8b5ea3255f50a3d14b0dd<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9799-8893c47.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9799-8893c47.dmg

**Comment by claudioshg** [comment]
Testing

**Comment by claudioshg** [comment]
### FAIL
Desktop PC - Windows 10 Home - CPU 4.0GHz Intel i7 - GTX1070 - RAM 16GB - Tested with Oculus Rift + Touch

Tested: Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9799-8893c47.exe

fail1pr9799
hifi-snap-by-claudioshg-on-2017-03-02_11-46-05

**NOTES**
- The lasers animate vertically. I can see at the far end the red dots, where I'm aiming, but the lasers animate incorrectly.

- I had one crash. I was testing the tablet response to the lasers. I was clicking on every icon, clockwise, from top to bottom. When I attempted to point and click, with my right hand the USERS icon, that's when it crashed.

- The jitter is very minimal now. It looks very nice. When I point and click at nothing, the lasers animate upwards. However, when I point and click on an entity, and I successfully far grab an entity, the laser animation corrects itself and I get the standard pink laser. The lasers jitter reduction is a PASS for me.

hifi-snap-by-claudioshg-on-2017-03-02_11-49-01

**Comment by sethalves** [comment]
yeah this was a bad merge.  I'm sorry I let you test that -- I'll fix it shortly

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9b3c16d489902e1f46426d3125420b4e7cbaa588 available at: https://gist.github.com/177554542bd8b5ea3255f50a3d14b0dd<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9799-9b3c16d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9799-9b3c16d.dmg

File: interface/src/ui/overlays/Line3DOverlay.cpp [hunk 1]
```diff
@@ -90,18 +122,19 @@ void Line3DOverlay::render(RenderArgs* args) {
     glm::vec4 colorv4(color.red / MAX_COLOR, color.green / MAX_COLOR, color.blue / MAX_COLOR, alpha);
     auto batch = args->_batch;
     if (batch) {
-        batch->setModelTransform(getTransform());
+        // batch->setModelTransform(getTransform());
```
Comment (ZappoMan) [inline interface/src/ui/overlays/Line3DOverlay.cpp]: dead code? remove

Comment (sethalves) [inline interface/src/ui/overlays/Line3DOverlay.cpp]: I'm not actually sure.  I would expect to have this line in place and use the local positions in the calls below, but that fails.  I need to ask an adult.

**Comment by claudioshg** [comment]
### FAIL

Tested: Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9799-9b3c16d.exe

**Re-Test NOTES**:
- In LOCALHOST, the laser disappears whenever I point it at the sky or in this example, the wall of the farthest room.
hifi-snap-by-claudioshg-on-2017-03-02_13-49-42
hifi-snap-by-claudioshg-on-2017-03-02_13-51-35

- The laser appears whenever I point at the ground, rocks, trees, building, except the sky or wall inside the building.
hifi-snap-by-claudioshg-on-2017-03-02_13-50-17

- In this example, the lasers are not visible when pointing at the sky, but reappear when I aim at the wall.
hifi-snap-by-claudioshg-on-2017-03-02_13-56-01

- The lasers on the tablet are also buggy. Activating the lasers on the tablet are sporadic. Sometimes they animate and sometimes they don't. More times they do, but here and there the lasers don't work.

- Distance grabbing's reduced jitter is still good.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5f570c1ba85d93f508587a30fa53de8f4b1e1532 available at: https://gist.github.com/177554542bd8b5ea3255f50a3d14b0dd<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9799-5f570c1.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9799-5f570c1.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 88d6fe6a0421e1d252f85114af338a039588363f available at: https://gist.github.com/177554542bd8b5ea3255f50a3d14b0dd<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9799-88d6fe6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9799-88d6fe6.dmg

**Comment by claudioshg** [comment]
### PASS

Tested: Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9799-88d6fe6.exe

- Everything works now. The lasers are immaculate. The lasers animate everywhere they are meant to work. Didn't find any fault using the lasers on the tablet or on entities. 
- The jitter is also to a very minimal. Very noticeable.

NOTES:
- The jitter seems to be at a minimal, but only when I am grabbing on to an entity. When I am not grabbing an entity, moving my hand rapidly left and right, I see many lasers. Note sure how important this is, but just making a note about it.
In the example below imagine 10 lasers, in a fan formation, and that's what I see. The gif can't capture exactly what I see.
The reason I bring this up is that the jitter for the lasers when not grabbing an entity is high. It's not a smooth laser moving left and right.
hifi-snap-by-claudioshg-on-2017-03-02_17-34-33

**Pull request merged**